server/indexer_service.go

```diff
@@ -54,6 +56,14 @@ func NewEVMIndexerService(
 // and indexing them by events.
 func (eis *EVMIndexerService) OnStart() error {
 	ctx := context.Background()
+
+	// when kava in state-sync mode, it returns zero as latest_block_height, which leads to undesired behavior, more
+	// details here: https://kava-labs.atlassian.net/wiki/spaces/ENG/pages/1623687169/EVM-Indexer+State+Sync+Issue
+	// to prevent this we wait until state-sync will finish
+	if err := waitUntilClientReady(ctx, eis.client, backoff.NewConstantBackOff(time.Second)); err != nil {
```

Comment:
is the default 15 minute maximum runtime long enough? i think in most cases it should be ok, but i've seen statesync take about that or longer. we may want to increase the default.

in any case, could be worth making that default explicit so someone could easily grok that we've thought about the case where the chain continues to return a latest height of 0 for a long time

Reply:
good catch, but it looks like `DefaultMaxElapsedTime` is applied only for `ExponentialBackOff` and is NOT applied for `ConstantBackOff`?

Reply:
@pirtleshell I can mention in the comment that `ExponentialBackOff` has `DefaultMaxElapsedTime`, but it can confuse?

Reply:
ooh, i think it'd be best to have a timeout mechanism. whether that is using `select` with a timer here or setting up exponential backoff with a shorter `MaxInterval` (1 min rechecking seems too long).

Reply:
fixed with `ExponentialBackOff` approach